pkg/index/fetch.go

```diff
@@ -17,33 +17,47 @@ package index
 import (
 	"context"
 	"fmt"
-	"io"
-	"net/http"
+	"net/url"
+	"strings"
 
+	"github.com/falcosecurity/falcoctl/pkg/index/fetch/http"
 	"gopkg.in/yaml.v3"
 )
 
-// Fetch retrieves a remote index using its URL.
-func Fetch(ctx context.Context, url, name string) (*Index, error) {
-	req, err := http.NewRequestWithContext(ctx, "GET", url, http.NoBody)
-	if err != nil {
-		return nil, fmt.Errorf("cannot fetch index: %w", err)
+type FetchFunc func(context.Context, string) ([]byte, error)
```

Comment:
For now I set the return signature to `[]byte` to keep the unmarshall logic in the main fetch function. Incase it is desired to support backends that store indices not as single YAML file we could also move the unmarshal logic into a util function and have the `Fetch` functions return an index entry list instead.